src/common/components/fileUploadButton/FileUploadButton.tsx

```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import { IconUpload } from "hds-react";
+
+import { ButtonAddFilesProps } from "../../../domain/feedbackForm/types";
+import styles from "./fileUploadButton.module.scss";
+
+// In order to provide an accessible label to the file upload while also
+// keeping in line with the design, I have to provide a button-like
+// element within the label. I can't do this with hds-react and I can't
+// use hds-core directly, because I don't want to create a version
+// conflict between HDS versions. The element can't be a button.
+//
+// To circumvent these limitation, I've included hds-core as a dev
+// dependency. This means that the version of HDS we use is integrated
+// during build time. This way we don't end up changing HDS style rules.
+// We may end up changing CSS variables.
+function FileUploadButton({ children, ...rest }: ButtonAddFilesProps) {
```

Comment:
If you think other projects will need the same types of buttons, I think we should create an issue to HDS (and maybe even a pull request if we have time)

Reply:
By the way: That issue to HDS shouldn't block this one from being merged, we can refactor this if HDS changed so that it isn't needed any more.

Reply:
I would guess that the need arises in conjunction with these sort of niche cases where you are in essence hacking to get what you want :D For instance here, we have the `<input type="file" />` which is rather difficult to style and flimsy to use.

From that perspective, my guess is that HDS would rather like to create implementations for these more difficult usecases. For instance in this case, they would maybe want to implement the `<FileInput />` itself. In that way they circumvent the need for a special button which is easy to use wrong semantically all the while implementing the difficult patterns in a single place and avoiding accessibility quirks with more confidence.